python/tests/spark/sql/codegen/test_mlflow_registry.py

```diff
@@ -0,0 +1,104 @@
+#  Copyright (c) 2021 Rikai Authors
+#
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+import os
+import sqlite3
+import uuid
+
+import mlflow
+import pytest
+import torch
+from mlflow.tracking import MlflowClient
+from pyspark.sql import Row, SparkSession
+from utils import check_ml_predict
+
+from rikai.spark.sql.codegen.mlflow_registry import log_model, MlflowModelSpec
+from rikai.spark.sql.schema import parse_schema
+
+
+@pytest.fixture(scope="module")
+def mlflow_client(
+    tmp_path_factory, resnet_model_uri: str, spark: SparkSession
+):
+    tmp_path = tmp_path_factory.mktemp("mlflow")
+    tmp_path.mkdir(parents=True, exist_ok=True)
+    tracking_uri = "sqlite:///" + str(tmp_path / "tracking.db")
+    mlflow.set_tracking_uri(tracking_uri)
+    experiment_id = mlflow.create_experiment("rikai-test", str(tmp_path))
+    # simpliest
+    with mlflow.start_run(experiment_id=experiment_id):
+        mlflow.log_param("optimizer", "Adam")
+        # Fake training loop
+        model = torch.load(resnet_model_uri)
+        artifact_path = "model"
+
+        schema = (
+            "struct<boxes:array<array<float>>,"
+            "scores:array<float>, labels:array<int>>"
+        )
+        pre_processing = (
+            "rikai.contrib.torch.transforms."
+            "fasterrcnn_resnet50_fpn.pre_processing"
+        )
+        post_processing = (
+            "rikai.contrib.torch.transforms."
+            "fasterrcnn_resnet50_fpn.post_processing"
+        )
+        log_model(
+            model,
+            artifact_path,
+            "pytorch",
```

Comment:
does this mean that it can only support the mlflow models where these tags exist?

Do we want to support vanilla mlflow models?

Reply:
if we do then we'd have to pass that extra information into CREATE MODEL or ML_PREDICT somehow. We can add support for that later on i just figured version 1 we keep it simple